### PR TITLE
Change APK base name depending on the branch name for debug builds

### PR DESCRIPTION
**Pull request by TobiGr**
#### What is it?
- [ ] Bug fix
- [ ] Feature
- [x] Development improvement

#### Long description of the changes in your PR

Debug APKs are now named after their branch name as suggested by @opusforlife2  e.g. the debug APK for this branch is named `NewPipe_apk_base_name-debug.apk`
Release builds are not touched, as well as debug builds on dev and master branch.

#### Testing apk

not needed

#### Agreement
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -46,6 +46,7 @@ android {
             } else {
                 applicationIdSuffix ".debug." + normalizedWorkingBranch
                 resValue "string", "app_name", "NewPipe " + workingBranch
+                archivesBaseName = 'NewPipe_' + workingBranch
```
Comment (Stypox) [inline app/build.gradle]: Better to use `normalizedWorkingBranch`, as `workingBranch` could contain special characters not allowed in filenames.
```suggestion
                archivesBaseName = 'NewPipe_' + normalizedWorkingBranch
```

**Comment by Stypox** [comment]
@TobiGr am I supposed to re-review this for some reason?

**Comment by TobiGr** [comment]
@Stypox no.

**Comment by Bubu** [comment]
This is somehow not working as intended. Even when doing a release build the apk is now called:
`NewPipe_head-release-unsigned.apk`. Also a file in the `META-INF` dir now ends up being called `NewPipe_head_release.kotlin_module` instead of `app_release.kotlin_module`. This is somehow not the case in you 0.19.2 github release build, so this breaks any chance for reproducibility unfortunately. 

And how do you generate the github release build? How does this not end up containing the kotlin_module rename?

I run `./gradlew clean assembleRelease`.

**Comment by TobiGr** [comment]
@B0pol Thanks for notifying us! I tested my changes locally when creating the PR and did not notice and changes in the release builds. Tbh. I have no idea why release builds are effected by this change. I can try to force the old naming of the release by adding `archivesBaseName = app` to the release part of the Gradle config.

> And how do you generate the github release build? How does this not end up containing the kotlin_module rename?

I build the app through Android studio by first doing `clean project` which is the equivalent of `./gradlew clean` and then build the unsigned app `Build > Build Bundle(s) / APK(s) > Build APK` which runs `./gradlew assembleRelease`. Signing is done in another step.

I

**Comment by Stypox** [comment]
I think you meant @Bubu, not @B0pol ;-)

**Pull request merged**